validmind/unit_metrics/__init__.py

```diff
@@ -0,0 +1,192 @@
+    else:
+        # If it's not a DataFrame, we cannot get a description
+        raise TypeError("The dataset provided is not a pandas DataFrame.")
+
+def _get_metric_class(metric_id):
+    """Get the metric class by metric_id
+
+    This function will load the metric class by metric_id.
+
+    Args:
+        metric_id (str): The full metric id (e.g. 'validmind.vm_models.test.v2.model_validation.sklearn.F1')
+
+    Returns:
+        Metric: The metric class
+    """
+
+    metric_module = importlib.import_module(f"{metric_id}")
+
+    class_name = metric_id.split(".")[-1]
+
+    # Access the class within the F1 module
+    metric_class = getattr(metric_module, class_name)
+
+    return metric_class
+
+def get_metric_value(metric_id, inputs=None):
+    """
+    Get the metric value, either by loading it from a global variable or by computing it,
+    ensuring that the inputs are the same.
+
+    Args:
+        metric_id (str): The metric name (e.g., 'F1').
+        inputs (dict, optional): Inputs necessary to compute the metric.
+
+    Returns:
+        The metric value.
+    """
+    # Serialize the inputs to ensure we can compare them
+    serialized_dataset = _serialize_dataset(inputs.get("dataset"))
+
+    # Use a tuple of the metric_id and the serialized inputs as the cache key
+    cache_key = (metric_id, serialized_dataset)
+
+    # Check if the metric value already exists in the global variable
+    if cache_key in global_metric_values:
+        print(f"Loading last computed value value from '{metric_id}'")
+        return global_metric_values[cache_key]
+    else:
+        # Compute the metric value
+        print(f"Computing metric value for '{metric_id}'")
+        result = run_metric(metric_id, inputs=inputs)
+        # Store the computed value in the global variable
+        global_metric_values[cache_key] = result
+        return result
+    
+def run_metric(metric_id, inputs=None, params=None):
+    """Run a single metric
+
+    This function provides a high level interface for running a single metric. A metric
+    is a single test that calculates a value based on the input data.
+
+    Args:
+        metric_id (str): The metric name (e.g. 'F1')
+        metric_value (dict): A dictionary of the metric value
+
+    Returns:
+        MetricResult: The metric result object
+    """
+
+    # Serialize the inputs to ensure we can compare them
+    serialized_dataset = _serialize_dataset(inputs.get("dataset"))
+
+    # Serialize the params to ensure we can compare them
+    serialized_params = _serialize_params(params)
+
+    serialized_model = _serialize_model(inputs.get("model"))
+
+    # Use a tuple of the metric_id and the serialized inputs as the cache key
+    cache_key = (metric_id, serialized_dataset, serialized_params, serialized_model)
+
+    # Check if the metric value already exists in the global variable
+    if cache_key in global_metric_values:
+        print(f"Loading last computed value value from '{metric_id}'")
```

Comment:
Let's use a `logger` instead so that we can control the logging verbosity. Once we start running more than 10 metrics at a time there will be too much output in the notebooks.

These `print()` calls can use `logger.debug()` so they are turned off by default (since the default logging level is `INFO`).